libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/ui/LocationComponent.kt

```diff
@@ -0,0 +1,83 @@
+package com.mapbox.navigation.ui.maps.internal.ui
+
+import android.animation.ValueAnimator
+import com.mapbox.geojson.Point
+import com.mapbox.maps.MapView
+import com.mapbox.maps.plugin.LocationPuck
+import com.mapbox.maps.plugin.locationcomponent.LocationConsumer
+import com.mapbox.maps.plugin.locationcomponent.location
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.internal.extensions.flowLocationMatcherResult
+import com.mapbox.navigation.ui.base.lifecycle.UIComponent
+import com.mapbox.navigation.ui.maps.location.NavigationLocationProvider
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlin.coroutines.resume
+
+@ExperimentalPreviewMapboxNavigationAPI
+class LocationComponent(
+    private val mapView: MapView,
+    private val locationPuck: LocationPuck,
+    private val locationProvider: NavigationLocationProvider,
```

Comment:
Android Auto uses the `CarAppLocation` as a source of truth. This can be accessed through 

```
MapboxNavigationApp.getObserver(..
```

Therefore does not need to be a dependency for the location component

Reply:
Recommending to consider moving `CarAppLocation` to the shared app. Otherwise AA can always transfer this while constructing a `LocationComponent`. So this one is not a requirement

Reply:
@tomaszrybakiewicz there is a goal to have a share car app location entity.

I'm looking to delete `MapboxCarApp` and one of the reasons it exists is to support `CarAppLocation`. If we can make it so AA depended on a service inside `SharedApp` then we could be a step closer to deleting `MapboxCarApp`

Reply:
Answered here https://github.com/mapbox/mapbox-navigation-android/pull/6206#discussion_r968832715